### PR TITLE
CDRIVER-2455 Fixed off-by-one error in BSON encoding of Regex

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

File: tests/test-bson.c [hunk 1]
```diff
@@ -2317,5 +2336,6 @@ test_bson_install (TestSuite *suite)
                   "/bson/unsupported_type/empty_key",
                   test_bson_visit_unsupported_type_empty_key);
    TestSuite_Add (suite, "/bson/binary_subtype_2", test_bson_subtype_2);
+   TestSuite_Add (suite, "/bson/CDRIVER-2455", test_bson_CDRIVER_2455);
```
Comment (ajdavis) [inline tests/test-bson.c]: Let's name it like the others. Perhaps "/bson/regex_length" and test_bson_regex_lengths.

**Review by ajdavis — CHANGES_REQUESTED** [review]

File: tests/test-bson.c (hunk 1 above)
Comment (derickr) [reply tests/test-bson.c]: Done

**Comment by ajdavis** [comment]
LGTM. Please commit to r1.9 in the libbson repo and master in the mongo-c-driver repo.

**Comment by derickr** [comment]
Fixed in Git

**Pull request closed without merging**